clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -679,11 +680,13 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
                 }
 
                 List<ConsumerRecord<K, V>> parsed = new ArrayList<>();
-                for (LogEntry logEntry : partition.records) {
+                Iterator<LogEntry> deepEntries = partition.records.deepIterator();
```

Comment:
Btw, it's a bit of a shame to lose the enhanced foreach syntax. Is there a reason not to expose a `deepIterable()` method instead?